### ISSUE TITLE
Clasp push results in Google Drive error for all deployments of Web App

### ISSUE THREAD

**Issue by TrojanCoder** [google/clasp#176]
## Expected Behavior
Write code locally and then push it for server side testing (the Web App relies on Google so I can only test so much locally). 

## Actual Behavior
When I push and then navigate to any deployment url, including the special HEAD one that ends in /dev, I get a web page with the title "Google Drive Error" and the following message:

> Google Docs encountered an error. Please try reloading this page, or coming back to it in a few minutes.
> 
> To learn more about the Google Docs editors, please visit our help center.
> 
> We're sorry for the inconvenience.
> - The Google Docs Team 

In order to view any changes I have to reload the browser Script Editor and go to Publish > Deploy as Web App

## Steps to Reproduce the Problem

1. `clasp create`
2. `echo "<h1>Hello World</h1>" > index.html`
3. Create following js file via this script:
```
cat << EOF > test.js
function doGet() {
    console.log('Running doGet()')
    return HtmlService
      .createTemplateFromFile('index')
      .evaluate();
}

function include(filename) {
    return HtmlService.createHtmlOutputFromFile(filename)
      .getContent();
}
EOF
```
4. `clasp push`
5. `clasp deployments`
6. Attempt to open the HEAD deployment
7. Manually publish and then HEAD works

## Specifications

  - Node version: 6.11.4
  - Version: 1.3
  - OS: Linux

**Comment by grant** [comment]
In the `Steps to Reproduce the Problem` section, can you provide exact steps including a minimal example (i.e. code)? All clasp steps you took (`clasp deploy`?)

That would help me debug this easier.

**Comment by TrojanCoder** [comment]
I've updated the steps. I'm not including deploy issues because it seems like that's covered by [#63](https://github.com/google/clasp/issues/63). If it matters adding a version and deployment after clasp push has no effect, and this happens after manually deploying the first time (so the script editor says the project is published in the lower left).

**Comment by TrojanCoder** [comment]
Thanks for the comment in #63, I never did a `clasp `pull after deploying in the interface because I didn't think anything changed so every time I pushed the `appscript.json file was reset.

Maybe it would be a good idea to check for HTML files to identify a web app and if so on `clasp `deploy update the `appscript.json file. This could be done with arguments to clasp deploy, an input prompt, or based on defaults stored in `.clasp.json

**Comment by grant** [comment]
@TrojanCoder Sounds like a good idea.
PRs and design plans are welcome!

**Comment by takanakahiko** [comment]
There are cases that can't be covered by that plan.

example:

```js
function doGet(){
  return ContentService.createTextOutput('Hello GAS!');
}
```

In this case, you don't use html files in project.

**Comment by takanakahiko** [comment]
Has this issue been solved by #431 ?

**Comment by grant** [comment]
@takanakahiko I think so. Closing.
Re-open if this is still an issue.
